src/test/kotlin/app/LoxoneAppTest.kt

```diff
@@ -0,0 +1,203 @@
+            get { lastModified }.isEqualTo(LAST_MODIFIED)
+            get { miniserverInfo.name }.isEqualTo("ShowRoom")
+            get { rooms.size }.isEqualTo(3)
+            get { rooms.values.first() }.isA<Room>()
+            get { categories.size }.isEqualTo(3)
+            get { categories.values.first() }.isA<Category>()
+            get { controls.size }.isEqualTo(6)
+            get { controls.values.first() }.isA<AlarmControl>()
+        }
+    }
+
+    @Test
+    fun `should getControl by type`() {
+        val alarmControl = AlarmControl()
+        val room = Room(UUID_ROOM, "room")
+        val category = Category(UUID_CATEGORY, "category")
+
+        expectThat(
+            LoxoneApp(
+                LAST_MODIFIED,
+                mockkClass(MiniserverInfo::class),
+                mapOf(Pair(UUID_ROOM, room)),
+                mapOf(Pair(UUID_CATEGORY, category)),
+                mapOf(Pair(UUID_CONTROL, alarmControl))
+            )
+        ) {
+            get { getControl(AlarmControl::class.java) }.isEqualTo(alarmControl)
+        }
+    }
+
+    @Test
+    fun `should getControl by name and type`() {
+        val switchControl = SwitchControl()
+        switchControl.name = "SomeControl"
+        val room = Room(UUID_ROOM, "room")
+        val category = Category(UUID_CATEGORY, "category")
+
+        expectThat(
+            LoxoneApp(
+                LAST_MODIFIED,
+                mockkClass(MiniserverInfo::class),
+                mapOf(Pair(UUID_ROOM, room)),
+                mapOf(Pair(UUID_CATEGORY, category)),
+                mapOf(Pair(UUID_CONTROL, switchControl))
+            )
+        ) {
+            get { getControl("SomeControl", SwitchControl::class.java) }.isEqualTo(switchControl)
+        }
+    }
+
+    @Test
+    fun `should getControlsForRoom by room`() {
+        val config = readResource("app/LoxAPP3.json", LoxoneApp::class)
+        val roomUuid = LoxoneUuid("0f869a64-028d-0cc2-ffffd4c75dbaf53c")
+
+        expectThat(config) {
+            get { rooms[roomUuid]?.let { room -> getControlsForRoom(room).size } }.isNotNull().isEqualTo(4)
+        }
+    }
+
+    @Test
+    fun `should getControlsForCategory by category`() {
+        val config = readResource("app/LoxAPP3.json", LoxoneApp::class)
+        val categoryUuid = LoxoneUuid("0f869a64-0221-0b28-ffffd4c75dbaf53c")
+
+        expectThat(config) {
+            get { categories[categoryUuid]?.let { cat -> getControlsForCategory(cat).size } }.isNotNull().isEqualTo(2)
+        }
+    }
+
+    @Test
+    fun `should getControlsForRoomAndCategory by room and category`() {
+        val config = readResource("app/LoxAPP3.json", LoxoneApp::class)
+        val categoryUuid = LoxoneUuid("0f869a64-0221-0b28-ffffd4c75dbaf53c")
+        val roomUuid = LoxoneUuid("0f869a64-028d-0cc2-ffffd4c75dbaf53c")
+
+        expectThat(config) {
+            get {
+                categories[categoryUuid]?.let { cat ->
```

Comment:
since we assert rooms and categories elsewhere, let's move them as vals outside the assertion for better readability